plugins/changevein.cpp

```diff
@@ -24,6 +24,188 @@ DFHACK_PLUGIN("changevein");
+            edges[WEST] |= (event->tile_bitmask.bits[i] & 0b0000000000000001) << i;
+        }
+
+        corners[NORTHEAST] = edges[NORTH] & 0b1000000000000000;
+        corners[SOUTHEAST] = edges[SOUTH] & 0b1000000000000000;
+        corners[SOUTHWEST] = edges[SOUTH] & 0b0000000000000001;
+        corners[NORTHWEST] = edges[NORTH] & 0b0000000000000001;
+    }
+
+    void SmudgeBitmask()
+    {
+        for (uint8_t i = 0; i < 4; i++)
+        {
+            edges[i] |= (edges[i] << 1) | (edges[i] >> 1);
+        }
+    }
+
+    bool Merge(VeinEdgeBitmask& other)
+    {
+        if (other.mat != this->mat)
+            return false;
+
+        for (uint8_t i = 0; i < 4; i++)
+        {
+            this->edges[i] |= other.edges[i];
+            this->corners[i] |= other.corners[i];
+        }
+
+        return true;
+    }
+
+    bool Merge(df::block_square_event_mineralst* event)
+    {
+        VeinEdgeBitmask temp = VeinEdgeBitmask(event);
+        return this->Merge(temp);
+    }
+};
+
+static void ChangeSameBlockVeins(df::map_block* block, df::block_square_event_mineralst* event, VeinEdgeBitmask& mask, int32_t matIndex)
+{
+    for (size_t i = 0; i < block->block_events.size(); i++)
+    {
+        df::block_square_event* evt = block->block_events[i];
+        if (evt == event)
+            continue;
+        if (evt->getType() != block_square_event_type::mineral)
+            continue;
+
+        df::block_square_event_mineralst* cur = (df::block_square_event_mineralst*)evt;
+        if (cur->inorganic_mat == mask.mat)
+        {
+            for (uint8_t j = 0; j < 15; j++)
+            {
+                uint16_t bitmask = cur->tile_bitmask.bits[j] | cur->tile_bitmask.bits[std::max(j - 1, 0)] | cur->tile_bitmask.bits[std::min(j + 1, 15)];
+                bitmask |= (bitmask << 1) | (bitmask >> 1);
+
+                if (bitmask & event->tile_bitmask.bits[j])
+                {
+                    mask.Merge(cur);
+                    cur->inorganic_mat = matIndex;
+                    break;
+                }
+            }
+        }
+    }
+}
+
+static void ChangeSurroundingBlockVeins(df::map_block* block, VeinEdgeBitmask& mask, int32_t matIndex)
+{
+    if (!block)
+        return;
+
+    // For detecting diagonal connections
+    mask.SmudgeBitmask();
+
+    for (uint8_t i = 0; i < 4; i++)
+    {
+        const df::coord2d& offset = DIRECTION_OFFSETS[i];
+        const uint16_t edgeBitmask = mask.edges[i];
+        const uint8_t oppositeIndex = (i + 2) % 4;
+
+        if (!edgeBitmask)
+            continue;
+
+        df::coord pos = df::coord(block->map_pos.x + offset.x, block->map_pos.y + offset.y, block->map_pos.z);
+        df::map_block* newBlock = Maps::getTileBlock(pos);
+
+        if (!newBlock)
+            continue;
+
+        for (size_t j = 0; j < newBlock->block_events.size(); j++)
+        {
+            df::block_square_event* evt = newBlock->block_events[j];
```

Comment:
```suggestion
        for (auto evt : newBlock->block_events)
        {
```

Reply:
same for other similar for loops in this file. the original code was written before this syntax was available in C++, but there's no reason not to use the modern syntax